File: libs/eave-stdlib-py/src/eave/stdlib/github_api/operations/verify_installation.py
from typing import Unpack
from eave.stdlib import requests_util
from eave.stdlib.core_api.operations import BaseRequestBody, BaseResponseBody
from eave.stdlib.github_api.operations import GithubAppEndpoint, GithubAppEndpointConfiguration


class VerifyInstallation(GithubAppEndpoint):
    config = GithubAppEndpointConfiguration(
        path="/github/api/verify",
        auth_required=False,
        team_id_required=False,
    )

    class RequestBody(BaseRequestBody):
        code: str
        installation_id: str

    class ResponseBody(BaseResponseBody):
        pass

    @classmethod
    async def perform(cls, input: RequestBody, **kwargs: Unpack[requests_util.CommonRequestArgs]) -> ResponseBody:
        response = await requests_util.make_request(
            config=cls.config,
            input=input,
            **kwargs,
        )

        body = await cls.make_response(response, cls.ResponseBody)
        return body

File: libs/eave-stdlib-ts/src/github-api/operations/verify-installation.ts
import { RequestArgsOrigin, makeRequest } from "../../requests.js";
import { GithubAppEndpointConfiguration } from "./shared.js";

export type VerifyInstallationRequestBody = {
  code: string;
  installation_id: string;
};

export class VerifyInstallationOperation {
  static config = new GithubAppEndpointConfiguration({
    path: "/github/api/verify",
    authRequired: false,
    teamIdRequired: false,
  });

  static async perform(
    args: RequestArgsOrigin & {
      input: VerifyInstallationRequestBody;
    },
  ): Promise<void> {
    await makeRequest({
      config: this.config,
      ...args,
    });
  }
}
